src/SFML/Graphics/SpriteBatch.cpp

```diff
@@ -0,0 +1,266 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2022 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Graphics/Font.hpp>
+#include <SFML/Graphics/RenderStates.hpp>
+#include <SFML/Graphics/RenderTarget.hpp>
+#include <SFML/Graphics/SpriteBatch.hpp>
+#include <SFML/System/Err.hpp>
+
+#include <algorithm>
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+void SpriteBatch::begin(BatchMode batchMode, RenderTarget& renderTarget, const Shader* shader, const BlendMode& blendMode)
+{
+    if (m_batchActive)
+    {
+        err() << "A batch is already active.";
+        return;
+    }
+
+    m_batchMode    = batchMode;
+    m_blendMode    = blendMode;
+    m_shader       = shader;
+    m_transform    = Transform::Identity;
+    m_renderTarget = &renderTarget;
+    m_batchActive  = true;
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::draw(const Batchable& drawable, float depth)
+{
+    drawable.batch(*this, depth);
+}
+
+
+////////////////////////////////////////////////////////////
+void SpriteBatch::draw(const Vertex*    vertices,
+                       std::size_t      count,
+                       PrimitiveType    type,
+                       const Texture*   texture,
+                       const Transform& transform,
+                       float            depth)
+{
+    bool unsupportedType = type != PrimitiveType::TriangleFan && type != PrimitiveType::TriangleStrip &&
```

Comment:
Can this be marked `const`?

Reply:
`unsupportedType`? Is there any benefit from marking it as `const`?

Reply:
Just the general benefits of liberally using `const` on stack variables. Makes the code easier to read and more maintainable as it communicates to future developers that this variable is not and should not be mutated.